docs/language-influences.rst

```diff
@@ -0,0 +1,25 @@
+###################
+Language Influences
+###################
+
+The Solidity language is a `curly-bracket language <https://en.wikipedia.org/wiki/List_of_programming_languages_by_type#Curly-bracket_languages>`_
+and has been influenced and inspired by several well-known programming languages.
+
+Solidity is most profoundly influenced by C++, but also borrowed concepts from other languages like
+Python, JavaScript, and others.
+
+In the early days of the language, Solidity used to be partly influenced by JavaScript.
+This was due to function-level scoping of variables and the use of the keyword ``var``.
+The JavaScript influence was more and more reduced starting from version 0.4.0.
+Now, the main similarity to JavaScript is that functions are defined using the keyword
+``function``. Solidity also supports import syntax and semantics that
+are similar to those available in JavaScript. Besides those points, Solidity looks like
+most other curly-bracket languages and has no major JavaScript influence anymore.
+
+Another influence to Solidity was Python. Solidity's modifiers were added trying to model
+Python's decorators with a much more restricted functionality. Furthermore, multiple inheritance
+and the ``super`` keyword are taken from Python as well as the general value semantics of value
+and reference types.
+
+Last but not least, ``payable`` was inspired by an, we believe, `unnamed language <https://publications.lib.chalmers.se/records/fulltext/234939/234939.pdf>`_
```

Comment:
Not sure this is really relevant.

Reply:
I just took what was there in the PR and Chris had listed it. :)

Reply:
Yeah but we debated there too :)

Reply:
Ok so @chriseth what do you think? Keep or delete?

Reply:
Ok so @chriseth wdyt? Keep it or delete it?

Reply:
I'm fine with either.